### PR TITLE
Enable automatic discovery of global editor configs

### PR DESCRIPTION
**Pull request by chsienki**
Automatically discover any files that end with `.globalconfig`
Allow the user to disable global config discovery
Add and enable tests.

Fixes https://github.com/dotnet/roslyn/issues/34688

### CODE REVIEW

File: src/Compilers/Core/MSBuildTask/Microsoft.Managed.Core.targets [hunk 1]
```diff
@@ -79,11 +79,12 @@
     
     -->
   <ItemGroup>
-    <_AllDirectoriesAbove Include="@(Compile->GetPathsOfAllDirectoriesAbove())" Condition="'$(DiscoverEditorConfigFiles)' != 'false'" />
+    <_AllDirectoriesAbove Include="@(Compile->GetPathsOfAllDirectoriesAbove())" Condition="'$(DiscoverEditorConfigFiles)' != 'false' or '$(DiscoverGlobalConfigFiles)' != 'false'" />
     <!-- Work around a GetPathsOfAllDirectoriesAbove() bug where it can return multiple equivalent paths when the 
          compilation includes linked files with relative paths - https://github.com/microsoft/msbuild/issues/4392 -->
     <PotentialEditorConfigFiles Include="@(_AllDirectoriesAbove->'%(FullPath)'->Distinct()->Combine('.editorconfig'))" Condition="'$(DiscoverEditorConfigFiles)' != 'false'" />
-    <EditorConfigFiles Include="@(PotentialEditorConfigFiles->Exists())" Condition="'$(DiscoverEditorConfigFiles)' != 'false'" />
+    <PotentialEditorConfigFiles Include="@(_AllDirectoriesAbove->'%(FullPath)'->Distinct()->Combine('.globalconfig'))" Condition="'$(DiscoverGlobalConfigFiles)' != 'false'" />
```
Comment (mavasani) [inline src/Compilers/Core/MSBuildTask/Microsoft.Managed.Core.targets:86]: Should we consider using a different item name, say `GlobalConfigFiles` or `GlobalAnalyzerConfigFiles`, so the same list can be used by NuGet/SDK/tool based providers for explicitly adding global config files?

**Review by mavasani — APPROVED** [review]
Thanks!

File: src/Compilers/Core/MSBuildTask/Microsoft.Managed.Core.targets (hunk 1 above)
Comment (chsienki) [reply src/Compilers/Core/MSBuildTask/Microsoft.Managed.Core.targets]: Good idea. I added GlobalEditorConfigFiles as an item group that a user can add to, and added a test to verify.

Comment (mavasani) [reply src/Compilers/Core/MSBuildTask/Microsoft.Managed.Core.targets]: Don't we want to avoid the term `EditorConfig` for global configs?

**Comment by chsienki** [comment]
@dotnet/roslyn-compiler for review please

**Comment by jaredpar** [comment]
Where do we have a good description of what globalconfig files are and how they play into the editor config story?

**Comment by mavasani** [comment]
@jaredpar - @chsienki @jasonmalinowski and me had an offline discussion in this space and came up with https://github.com/dotnet/roslyn/issues/47707. This PR aims to address one of the confusion/pain points when end users want to author global configs.

**Comment by jaredpar** [comment]
Okay I see. This isn't really changing our implementation then, it's about the presentation of the feature. Want to push developers into naming global editor configs `.globalconfig` instead of `.editorconfig`. But the implementation doesn't really care about the name, it's just looking at the file format.

**Review by jaredpar — APPROVED** [review]

File: src/Compilers/Core/MSBuildTask/Microsoft.Managed.Core.targets (hunk 1 above)
Comment (chsienki) [reply src/Compilers/Core/MSBuildTask/Microsoft.Managed.Core.targets]: Hmm, good point will switch to `GlobalAnalyzerConfigFiles` don't want just globalconfig as that seems to general.

**Comment by chsienki** [comment]
@dotnet/roslyn-compiler for a second compiler review please :)

**Review by cston — APPROVED** [review]

**Pull request merged**